internal/mode/static/state/change_processor_test.go

```diff
@@ -1662,525 +1663,365 @@ var _ = Describe("ChangeProcessor", func() {
 				Expect(changed).To(BeFalse())
 			})
 		})
-		Describe("Multiple Kubernetes API resource changes", Ordered, func() {
```

Comment:
Unsure of how to refactor these tests as the capturer is now gone. Also, unsure of the goal of these tests and if they are necessary. Thoughts?

Reply:
Okay I added some tests back in, it was a little tricky as now we are always regenerating the config on these Gateway resources if we call upsert on them so some of the tests I couldn't really translate. e.g. expecting any upsert on a Gateway resource to be a non-changing change. I'm pretty unsure of what I wrote and if it does its job so please let me know of your thoughts!

Reply:
Looks good to me!

Reply:
> . I'm pretty unsure of what I wrote and if it does its job so please let me know of your thoughts!

the updated tests look reasonable.

However, if we have endpoints and slices there, shall we also add Secrets and Namespaces? This way, we will test a combination of all possibly related resources

@kate-osborn

Reply:
Yeah, we should. Thanks for catching @pleshakov

Reply:
added some changes, do you think they need any comments or are they fine as is?